### PR TITLE
RD-3434 accurate requests time

### PR DESCRIPTION
**Pull request by saartochner**
For the lambda:
image

Before:
image
After:
image

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -5,3 +5,4 @@ boto3
 pytest-cov==2.6.1
 capturer==2.4
 attrs==19.1.0
+requests
```
Comment (nirLumigo) [inline requirements.txt:8]: Why not pin the version?

File: src/lumigo_tracer/sync_http/sync_hook.py [hunk 2]
```diff
@@ -313,4 +328,6 @@ def wrap_http_calls():
                 wrap_function_wrapper(
                     "urllib3.response", "HTTPResponse.read_chunked", _read_stream_wrapper
                 )
+            if importlib.util.find_spec("requests"):
```
Comment (nirLumigo) [inline src/lumigo_tracer/sync_http/sync_hook.py:331]: 🚀

File: src/lumigo_tracer/sync_http/sync_hook.py [hunk 3]
```diff
@@ -313,4 +328,6 @@ def wrap_http_calls():
...
+                wrap_function_wrapper("requests.api", "request", _requests_wrapper)
```
Comment (nirLumigo) [inline src/lumigo_tracer/sync_http/sync_hook.py:332]: Won't there a double wrapping for the request?

File: src/test/unit/sync_http/test_sync_hook.py [hunk 4]
```diff
@@ -502,6 +503,27 @@ def lambda_test_function(event, context):
...
+        start_time = time.time() * 1000
+        requests.get("https://www.google.com")
+        return start_time
+
+    # add delay to the connection establishment process
+    original_getaddrinfo = socket.getaddrinfo
+    monkeypatch.setattr(
+        socket,
+        "getaddrinfo",
+        lambda *args, **kwargs: time.sleep(0.1) or original_getaddrinfo(*args, **kwargs),
```
Comment (nirLumigo) [inline src/test/unit/sync_http/test_sync_hook.py:518]: I think it should be extracted to a function, I don't think we should combine lines by putting `or` between them, makes it hard to read.

File: src/lumigo_tracer/sync_http/sync_hook.py (hunk 3 above)
Comment (saartochner) [reply src/lumigo_tracer/sync_http/sync_hook.py:332]: there is!
Note that `_requests_wrapper` (the "higher" hook) doesn't create span by itself, but only append data to the span that was created with `http.client` (the lower hook).

**Pull request merged**